### PR TITLE
Release v0.42.0

### PR DESCRIPTION
**Pull request by ljoss17**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,40 @@
 # CHANGELOG
 
+## v0.42.0
+
+*February 8th, 2024*
+
+This release updates the proto types from ibc-go from `v8.0.0` to `v8.1.0`.
```
Comment (romac) [inline CHANGELOG.md]:
```suggestion
This release updates the proto types for IBC-Go from `v8.0.0` to `v8.1.0`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+This includes proto types used for channel upgrade.
+
+Please note that ibc-go `v8.1.0` doesn't have the WASM light client proto types. These types will live in
```
Comment (romac) [inline CHANGELOG.md]:
```suggestion
Please note that IBC-Go v8.1.0 doesn't provide the Wasm light client proto types anymore (`ibc.lightclients.wasm.v1`). These types will live in
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+  valid UTF-8 data for its `key` and `value` attributes, specifying 
+  them as `bytes` instead of `string`. ([#180](https://github.com/cosmos/ibc-proto-rs/issues/180))
+
+  This is required, because ibc-go emits event attributes which are not valid UTF-8,
+  so we need to use this definition to be able to parse them.
+
+  In Protobuf, `bytes` and `string` are wire-compatible, so doing this strictly increases the amount fo data we can parse.
+
+  See this Hermes PR for background information: https://github.com/informalsystems/hermes/pull/3768
+- Removed WASM light client proto types
```
Comment (romac) [inline CHANGELOG.md]: Let's move this as the first item, and add a link to the v8.1.0 PR (192)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+
+### FEATURES
+
+- Bump ibc-go to v8.1.0 ([\#192](https://github.com/cosmos/ibc-proto-
+  rs/pull/192))
```
Comment (romac) [inline CHANGELOG.md]:
```suggestion
- Bump IBC-Go to v8.1.0 ([\#192](https://github.com/cosmos/ibc-proto-rs/pull/192))
```

**Review by romac — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (ljoss17) [reply CHANGELOG.md]: Done in eb6d3c87177b528a82ac950d96e6745cc51e85b8

**Review by romac — APPROVED** [review]

**Pull request merged**